src/main/java/org/opendatasoft/elasticsearch/ingest/GeoExtensionProcessor.java

```diff
@@ -265,10 +267,33 @@ public IngestDocument execute(IngestDocument ingestDocument) throws IOException,
             if (centroidField != null) ingestDocument.setFieldValue(
                     geoShapeField + "." + centroidField, GeoUtils.getCentroidFromGeom(geom));
             if (bboxField != null) {
-                Coordinate[] coords = geom.getEnvelope().getCoordinates();
-                if (coords.length >= 4) ingestDocument.setFieldValue(
-                        geoShapeField + "." + bboxField,
-                        GeoUtils.getBboxFromCoords(coords));
+                Coordinate[] coords;
+                if (geom.getGeometryType() == "Point") {
+                    coords = geom.getCoordinates();
+                    if (coords.length == 1) {
+                        GeoPoint topLeft = new GeoPoint(
+                                org.elasticsearch.common.geo.GeoUtils.normalizeLat(coords[0].y),
+                                org.elasticsearch.common.geo.GeoUtils.normalizeLon(coords[0].x)
+                        );
+                        GeoPoint bottomRight = new GeoPoint(
+                                org.elasticsearch.common.geo.GeoUtils.normalizeLat(coords[0].y),
+                                org.elasticsearch.common.geo.GeoUtils.normalizeLon(coords[0].x)
+                        );
+
+                        ingestDocument.setFieldValue(
+                                geoShapeField + "." + bboxField,
+                                Arrays.asList(topLeft, bottomRight)
+                        );
+                    }
+                } else {
+                    coords = geom.getEnvelope().getCoordinates();
```

Comment:
what is coords if `geom` is a point here ?

Reply:
coords = [(-3.3802714930640434, 47.755537652836495, NaN)]

Reply:
So then you could just test the length of coords for 1 and return the bbox based on this single point ?

Reply:
Do you want me to avoid the "if and else" and instead use the length of this result to create bbox ?

Reply:
Yes I think you can avoid the "point" string comparison at all like this